script/plugin_webpack_directories.rb

```diff
@@ -16,29 +16,61 @@
   end
 end
 
+def log_skip_plugin(plugin_name, reason)
+  STDERR.puts "Skipping plugin #{plugin_name}: #{reason}"
+end
+
+def log_plugin(plugin_name)
+  STDERR.puts "Processing plugin #{plugin_name}"
+end
+
+def entry_name(plugin_name, entry)
+  prefix = File.basename(entry).gsub(/[_]?index.js$/, '')
+  if prefix != ''
+    "#{plugin_name}_#{prefix}"
+  else
+    plugin_name
+  end
+end
+
 config = { entries: {}, paths: [], plugins: {} }
+entry_paths = []
 plugin_name_regexp = /foreman*|katello*/
 specs.each do |dep|
   # skip other rails engines that are not plugins
   # TODO: Consider using the plugin registration api?
   next unless dep.name =~ plugin_name_regexp
-  next if dep.name.include?('_core')
   dep = dep.to_spec if gemfile_in
 
   path = "#{dep.to_spec.full_gem_path}/webpack"
-  # some plugins share the same base directory (tasks-core and tasks, REX etc)
-  # skip the plugin if its path is already included
-  next if config[:paths].include?(path)
-  next unless Dir.exist?(path)
-  next unless File.exist?("#{dep.to_spec.full_gem_path}/package.json")
+
+  unless Dir.exist?(path)
+    log_skip_plugin(dep.name, './webpack directory missing')
+    next
+  end
+  unless File.exist?("#{dep.to_spec.full_gem_path}/package.json")
+    log_skip_plugin(dep.name, './package.json missing')
+    next
+  end
+  log_plugin(dep.name)
+
   package_json = JSON.parse(File.read("#{dep.to_spec.full_gem_path}/package.json"))
-  main = package_json['main'] || 'index.js'
-  entry = "#{path}/#{main}"
-  if File.exist?(entry)
-    config[:entries][dep.name] = entry
-    config[:paths] << path
-    config[:plugins][dep.name] = {root: dep.to_spec.full_gem_path, entry: entry}
+
+  entries = Dir["#{path}/*index.js"]
```

Comment:
Is this part of the spec? The previous implementation was based on https://docs.npmjs.com/files/package.json#main that I interpreted as you should use main if present.

Reply:
This extends the default npm behavior. The main entry point is still used if it's available (see the next line). On top of that all `*index,js` files are added which allows adding more independent entry points. I used that in Katello where I needed to load some npm dependencies into angular pages (see [common_index.js](https://github.com/Katello/katello/pull/7001/files#diff-106a751f2719195939e27fd323a88a12) in Katello).

Reply:
I'm hesitant to extend default behavior but would be OK with it if we can describe the interface we provide for plugins and which part goes beyond default npm behavior.